src/config.rs

```diff
@@ -12,35 +12,98 @@
 // along with this software.
 // If not, see <https://opensource.org/licenses/MIT>.
 
+use crate::Error;
+use farcaster_core::blockchain::Network;
 use internet2::NodeAddr;
-use lnpbp::chain::Chain;
+use std::path::Path;
+
+use serde::{Deserialize, Serialize};
+
+pub const FARCASTER_MAINNET_ELECTRUM_SERVER: &str = "ssl://blockstream.info:700";
+pub const FARCASTER_MAINNET_MONERO_DAEMON: &str = "http://node.monerooutreach.org:18081";
+pub const FARCASTER_MAINNET_MONERO_RPC_WALLET: &str = "http://localhost:18083";
+
+pub const FARCASTER_TESTNET_ELECTRUM_SERVER: &str = "ssl://blockstream.info:993";
+pub const FARCASTER_TESTNET_MONERO_DAEMON: &str = "http://stagenet.melo.tools:38081";
+pub const FARCASTER_TESTNET_MONERO_RPC_WALLET: &str = "http://localhost:38083";
 
 #[cfg(feature = "shell")]
 use crate::opts::Opts;
 
-/// Final configuration resulting from data contained in config file environment
-/// variables and command-line options. For security reasons node key is kept
-/// separately.
-#[derive(Clone, PartialEq, Eq, Debug, Display)]
-#[display(Debug)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(crate = "serde_crate")]
 pub struct Config {
-    /// Bitcoin blockchain to use (mainnet, testnet, signet, liquid etc)
-    pub chain: Chain,
-
-    /// ZMQ socket for lightning peer network message bus
-    pub msg_endpoint: NodeAddr,
+    /// Syncer configuration
+    pub syncers: Option<Syncers>,
+}
 
-    /// ZMQ socket for internal service control bus
-    pub ctl_endpoint: NodeAddr,
+impl Config {
+    pub fn get_syncer_servers(&self, network: Network) -> Option<SyncerServers> {
+        match network {
+            Network::Mainnet => self.syncers.as_ref()?.mainnet.clone(),
+            Network::Testnet => self.syncers.as_ref()?.testnet.clone(),
+            Network::Local => None,
+        }
+    }
 }
 
-#[cfg(feature = "shell")]
-impl From<Opts> for Config {
-    fn from(opts: Opts) -> Self {
+impl Default for Config {
+    fn default() -> Self {
         Config {
-            chain: opts.chain,
-            msg_endpoint: opts.msg_socket.into(),
-            ctl_endpoint: opts.ctl_socket.into(),
+            syncers: Some(Syncers::default()),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(crate = "serde_crate")]
+pub struct Syncers {
```

Comment:
Syncere is too generic a name for my taste, especially since we have other syncer vars in the code. How about just `SyncerConfig`

Reply:
Done in 9e3a5fef5cc0610d9b79137e86706d6a1b7ff378